jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -740,6 +737,8 @@ ecma_string_to_utf8_string (const ecma_string_t *string_desc_p, /**< ecma-string
   }
 
   JERRY_ASSERT (size <= buffer_size);
+  JERRY_ASSERT (size == ecma_string_get_size (string_desc_p));
```

Comment:
I think this is checked at the beginning of the function.

Personally I would add another always inline function, which does not have a return value, and checks whether the size is the same as the string length, and use it whenever possible.

The name would be: ecma_string_to_utf8_bytes - it emphasises that we want the strinmg representation and nothing else
And the current function would be renamed to ecma_string_copy_to_utf8_buffer - it emphasises that we have a generic buffer where we combine strings

The first one would be an always inline function without return value, which calls the second. But before you do this change, I would like to hear the opinion of other reviewers.